packages/theia-integration/src/browser/diagram/glsp-theia-diagram-configuration.ts

```diff
@@ -61,7 +61,9 @@ export abstract class GLSPTheiaDiagramConfiguration implements DiagramConfigurat
         container.bind(OpenerService).toConstantValue(this.openerService);
         container.bind(CommandService).toConstantValue(this.commandService);
         container.bind(ExternalModelSourceChangedHandler).toConstantValue(this.modelSourceChangedHandler);
-        container.rebind(CommandPalette).to(TheiaCommandPalette);
+        if (container.isBound(CommandPalette)) {
+            container.rebind(CommandPalette).to(TheiaCommandPalette);
+        }
```

Comment:
Looks good to me but shouldn't we do a "normal" bind if it is not already bound?

Reply:
Agreed

Reply:
No, I think it doesn't really make sense to bind it if the `CommandPalette` isn't bound. If it isn't bound, the diagram editor doesn't want to have one, so we shouldn't then bind the `TheiaCommandPalette` either. Wdyt?

Reply:
Ah right, sorry, I somehow was not thinking about that! You are absolutely right!